test/integration/login-otp-email.e2e-spec.ts

```diff
@@ -0,0 +1,102 @@
+import * as request from "supertest";
+
+import { DataSource, Repository } from "typeorm";
+import { INestApplication, ValidationPipe } from "@nestjs/common";
+import { Otp, User } from "../../src/entities";
+import { OtpFixture, UserFixture } from "../fixtures";
+import { Test, TestingModule } from "@nestjs/testing";
+
+import { AppModule } from "../../src/app.module";
+import { LoginOtpEmailRequest } from "../../src/login/dto/login-otp-email-request.dto";
+import { MockFactory } from "mockingbird";
+import { faker } from "@faker-js/faker";
+import { setupTestDataSourceAsync } from "../test-db";
+
+describe("LoginOtpEmailController (e2e)", () => {
+  let app: INestApplication;
+  let moduleFixture: TestingModule;
+  let userRepository: Repository<User>;
+  let otpRepository: Repository<Otp>;
+
+  beforeEach(async () => {
+    moduleFixture = await Test.createTestingModule({
+      imports: [AppModule],
+    })
+      .overrideProvider(DataSource)
+      .useValue(await setupTestDataSourceAsync())
+      .compile();
+
+    app = moduleFixture.createNestApplication();
+    app.useGlobalPipes(new ValidationPipe());
+    await app.init();
+    userRepository = moduleFixture.get<Repository<User>>("UserRepository");
+    otpRepository = moduleFixture.get<Repository<Otp>>("OtpRepository");
+  });
+
+  afterEach(async () => {
+    await app.close();
+    await moduleFixture.close();
+  });
+
+  describe("POST /login-otp-email", () => {
+    it("Should return a token if OTP credentials are valid", async () => {
+      const mockUser = MockFactory(UserFixture).one().hashPassword();
```

Comment:
General note for namings. Do not name as mockUser, mockOtp etc. Define the variable well. Which user is it? Which otp is it? Name it well for better understanding the test.